mm2src/crypto/src/crypto_ctx.rs

```diff
@@ -336,6 +336,18 @@ impl CryptoCtx {
     }
 }
 
+#[macro_export]
+macro_rules! mm2_internal_pubkey_hex {
+    ($ctx: ident, $err_construct: expr) => {
+        match CryptoCtx::from_ctx(&$ctx).split_mm() {
+            Ok(crypto_ctx) => Ok(Some(crypto_ctx.mm2_internal_pubkey_hex())),
+            Err((CryptoCtxError::NotInitialized, _)) => Ok(None),
+            Err((CryptoCtxError::Internal(error), trace)) => MmError::err_with_trace($err_construct(error), trace),
+        }
+    };
+}
+pub use mm2_internal_pubkey_hex;
```

Comment:
Why change the function to a macro? a function is always preferred whenever possible.

Reply:
The only reason I used a macro was my wrong belief regarding using enum field as a method. Certainly we should use macro only as a syntax sugar, it's my fault.
The other thing is where it should be, now I think `crypto_ctx.rs` is wrong that is why and I moved it into `mm2_main/mm2.rs`. Thank you

[done](https://github.com/KomodoPlatform/atomicDEX-API/pull/1849/commits/c5d071708a7ce16f847b5505733c75ac1cf35a16)

Reply:
> The other thing is where it should be, now I think crypto_ctx.rs is wrong that is why and I moved it into mm2_main/mm2.rs.

I think the best place would be `lp_ordermatch.rs` since this function is only used inside `lp_ordermatch` folder

Reply:
[done](https://github.com/KomodoPlatform/atomicDEX-API/pull/1849/commits)